Source/Dafny/Compilers/Compiler-Csharp.cs

```diff
@@ -3207,5 +3291,259 @@ private class CSharpCompilationResult {
       wBody.WriteLine($"{GetHelperModuleName()}.WithHaltHandling({companion}.{idName});");
       Coverage.EmitTearDown(wBody);
     }
+
+    /// <summary>
+    /// Below is the full grammar of ensures clauses that can specify
+    /// the behavior of an object returned by a mock-annotated method
+    /// (S is the starting symbol, ID refers to a variable/field/method/type
+    /// identifier, EXP stands for an arbitrary Dafny expression):
+    ///
+    /// S       = FORALL
+    ///         | EQUALS 
+    ///         | S && S
+    /// 
+    /// EQUALS  = ID.ID (ARGLIST)  == EXP // stubs a method call
+    ///         | ID.ID            == EXP // stubs field access
+    ///         | EQUALS && EQUALS
+    /// 
+    /// FORALL  = forall SKOLEMS :: EXP ==> EQUALS
+    /// 
+    /// ARGLIST = ID  // this can be one of the skolem variables
+    ///         | EXP // this exp may not reference any of the skolems 
+    ///         | ARG_LIST, ARG_LIST
+    /// 
+    /// SKOLEMS = ID : ID 
+    ///         | SKOLEMS, SKOLEMS
+    /// 
+    /// </summary>
+    private class MockWriter {
+
+      private readonly CsharpCompiler compiler;
+      // maps identifier names to the names of corresponding mocks:
+      private Dictionary<string, string> mockNames = new();
+      // associates a skolem variable with the lambda passed to argument matcher
+      private List<Tuple<IVariable, string>> bounds = new();
+
+      public MockWriter(CsharpCompiler compiler) {
+        this.compiler = compiler;
+      }
+
+      /// <summary>
+      /// Create a body for a method returning a fresh instance of an object 
+      /// </summary>
+      public ConcreteSyntaxTree CreateFreshMethod(Method m,
```

Comment:
Please use fully written out variable names in the general case, so `method` here.

Reply:
Made this change for my code. `m` seems to be used instead of `method` throughout the compilers - should I change the other instances as well?